_resources/port1.0/group/app-1.0.tcl

```diff
@@ -218,18 +218,24 @@ platform macosx {
                 return -code error "app.executable ${app.executable} should not start with \${destroot}"
             }
 
-            # If app.executable is in the destroot, link to it.
+            # If app.executable is in the destroot, write a script to launch it.
             if {[file exists ${destroot}[app._resolve_symlink ${executable} ${destroot}]]} {
-                ln -s ${executable} ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}
+                set launch_script [open ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name} w]
+                puts ${launch_script} "#!/bin/bash\n${executable}\n"
+                close ${launch_script}
+                system -W ${worksrcpath} "/bin/chmod 755 ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}"
             } elseif {[file exists ${executable}]} {
                 # If app.executable starts with ${workpath} or ${filespath}, copy it.
                 if {[string first ${workpath} ${executable}] == 0 || [string first ${filespath} ${executable}] == 0} {
                     xinstall ${executable} ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}
                 
                 # app.executable refers to a file that exists but does not belong to this port.
-                # Assume it belongs to a dependency and symlink it.
+                # Assume it belongs to a dependency and write a script to launch it.
                 } else {
-                    ln -s ${executable} ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}
+                    set launch_script [open ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name} w]
+                    puts ${launch_script} "#!/bin/bash\n${executable}\n"
+                    close ${launch_script}
+                    system -W ${worksrcpath} "/bin/chmod 755 ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}"
```

Comment:
Ditto. The code repetition might be a reason to make a `proc` for this.

Reply:
Agreed - I tried to make it a proc for a couple of hours, but all the various combinations and permutations failed for me, so I gave in and copy/pasted the code. It was that system call I was getting stuck at - just plain failed to run inside a proc for some reason; maybe the tcl functions would work better for me.